packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverPageCount/renderTeachingPopoverPageCount.tsx

```diff
@@ -1,15 +1,69 @@
-/** @jsxRuntime automatic */
-/** @jsxImportSource @fluentui/react-jsx-runtime */
-
-import { assertSlots } from '@fluentui/react-utilities';
-import type { TeachingPopoverPageCountState, TeachingPopoverPageCountSlots } from './TeachingPopoverPageCount.types';
+import * as React from 'react';
+import type { TeachingPopoverPageCountState } from './TeachingPopoverPageCount.types';
+import { getSlotsNext } from '@fluentui/react-utilities';
+import { TeachingPopoverPageCountSlots } from './TeachingPopoverPageCount.types';
 
 /**
  * Render the final JSX of TeachingPopoverPageCount
  */
 export const renderTeachingPopoverPageCount_unstable = (state: TeachingPopoverPageCountState) => {
-  assertSlots<TeachingPopoverPageCountSlots>(state);
+  const { slots, slotProps } = getSlotsNext<TeachingPopoverPageCountSlots>(state);
+
+  const { setCurrentPage, currentPage, totalPages, countStyle } = state;
+
+  if (totalPages <= 1) {
+    // We don't need pagination for one or zero pages
+    return null;
+  }
+
+  if (countStyle === 'text') {
+    // Start index of 1
+    // Users pass localized text as child (i.e. 'of').
```

Comment:
We should review this with some localization folks as I'm not sure this is sufficient for localization.